test/server/options_impl_test.cc

```diff
@@ -29,11 +29,12 @@ TEST(OptionsImplDeathTest, HotRestartVersion) {
 
 TEST(OptionsImplTest, All) {
   std::unique_ptr<OptionsImpl> options = createOptionsImpl(
-      "envoy --concurrency 2 -c hello --restart-epoch 1 -l info "
+      "envoy --concurrency 2 -c hello -a path --restart-epoch 1 -l info "
       "--service-cluster cluster --service-node node --service-zone zone "
       "--file-flush-interval-msec 9000 --drain-time-s 60 --parent-shutdown-time-s 90");
   EXPECT_EQ(2U, options->concurrency());
   EXPECT_EQ("hello", options->configPath());
+  EXPECT_EQ("path", options->adminAddressPath());
   EXPECT_EQ(1U, options->restartEpoch());
   EXPECT_EQ(spdlog::level::info, options->logLevel());
   EXPECT_EQ("cluster", options->serviceClusterName());
```

Comment:
Can you also add a test somewhere to make sure the admin address file contains the expected address?